tests/h/presenters/annotation_json_test.py

```diff
@@ -137,26 +116,25 @@ def test_immutable_formatters(self, groupfinder_service, links_service):
         """
         ann = mock.Mock(id="the-real-id", extra={})
         resource = AnnotationContext(ann, groupfinder_service, links_service)
+        formatters = []
 
-        formatters = [FakeFormatter({"flagged": "nope"})]
         presenter = AnnotationJSONPresenter(resource, formatters)
-        formatters.append(FakeFormatter({"enterprise": "synergy"}))
+        formatters.append(get_formatter({"enterprise": "synergy"}))
         presented = presenter.asdict()
 
         assert "enterprise" not in presented
 
     def test_formatter_uses_annotation_resource(
-        self, groupfinder_service, links_service
+        self, groupfinder_service, links_service, get_formatter
     ):
         annotation = mock.Mock(id="the-id", extra={})
         resource = AnnotationContext(annotation, groupfinder_service, links_service)
+        formatter = get_formatter()
 
-        formatters = [IDDuplicatingFormatter()]
-        presenter = AnnotationJSONPresenter(resource, formatters)
-
-        output = presenter.asdict()
+        presenter = AnnotationJSONPresenter(resource, formatters=[formatter])
+        presenter.asdict()
 
-        assert output["duplicated-id"] == "the-id"
+        formatter.format.assert_called_once_with(resource)
```

Comment:
It was in comments from the `IDDuplicatingFormatter` but the whole reason that was done was to indirectly prove we were called with the context. Using a mock we can just assert it

Reply:
:+1: